fiboa_cli/publish.py

```diff
@@ -0,0 +1,97 @@
+
+    parent = os.path.dirname(directory)
+    os.chdir(parent)
+
+    if not source_coop_extension:
+        source_coop_extension = dataset.replace("_", "-")
+
+    file_name = source_coop_extension.replace("-", "_")  # not sure if we want this
+
+    parquet_file = os.path.join(directory, f"{file_name}.parquet")
+    source_coop_url = f"https://beta.source.coop/fiboa/{source_coop_extension}/"
+
+    assert requests.get(f"https://api.source.coop/repositories/fiboa/{source_coop_extension}").status_code == 200, \
+        f"Missing repo at {source_coop_url}"
+
+    collection_file = os.path.join(directory, "collection.json")
+
+    stac_directory = os.path.join(directory, "stac")
+    done_convert = os.path.exists(parquet_file) and os.path.exists(os.path.join(stac_directory, "collection.json"))
+
+    if not done_convert:
+        # fiboa convert xx_yy -o data/xx-yy.parquet -h https://beta.source.coop/fiboa/xx-yy/ --collection
+        log(f"Converting file for {dataset} to {parquet_file}\n", "success")
+        convert(dataset, parquet_file, cache, source_coop_url, collection=True)
+        log(f"Done\n", "success")
+    else:
+        log(f"Using existing file {parquet_file} for {dataset}\n", "success")
+
+    # fiboa validate data/xx-yy.parquet --data
+    log(f"Validating {parquet_file}", "info")
+    result = validate(parquet_file, config={"data": True})
+    if result:
+        log("\n  => VALID\n", "success")
+    else:
+        log("\n  => INVALID\n", "error")
+        sys.exit(1)
+
+    # mkdir data/stac; mv data/collection.json data/stac
+    stac_target = os.path.join(stac_directory, "collection.json")
+    if not done_convert:
+        os.makedirs(stac_directory, exist_ok=True)
+        data = json.load(open(collection_file))
+        assert data["id"] == dataset, f"Wrong collection dataset id: {data['id']} != {dataset}"
+
+        if STAC_EXTENSION not in data["stac_extensions"]:
+            data["stac_extensions"].append(STAC_EXTENSION)
+            data["links"].append({
+                "href": f"{source_coop_url}/{file_name}.pmtiles",
+                "type": "application/vnd.pmtiles",
+                "rel": "pmtiles"
+            })
+
+        with open(stac_target, "w", encoding="utf-8") as f:
+            json.dump(data, f)
+        os.remove(collection_file)
+
+    pm_file = os.path.join(directory, f"{file_name}.pmtiles")
+    if not os.path.exists(pm_file):
+        log(f"ogr2ogr geo.json", "info")
+        exc(f"ogr2ogr -t_srs EPSG:4326 geo.json {parquet_file}")
+
+        log(f"Running tippecanoe", "info")
+        exc(f"tippecanoe -zg --projection=EPSG:4326 -o {pm_file} -l {dataset} geo.json --drop-densest-as-needed")
+
+        os.remove("geo.json")
+
+    log(f"Uploading to aws", "info")
+    log(f"Get your credentials at {source_coop_url}download/ (press ACCESS DATA and GENERATE CREDENTIALS", "info")
+    exc(f"aws s3 sync {directory} s3://us-west-2.opendata.source.coop/fiboa/{source_coop_extension}/")
```

Comment:
I'm just wondering, but shouldn't the CLI stop after the log and let users input the credentials?

Also, we are backing away from using the fiboa org and aim at users hosting under their own accounts, so this may need some slight changes.

Reply:
Only if it fails. After generating credentials, source coop shows a pre-filled _Usage Examples_ section with export-statements. Copy-paste in your terminal and re-run the publish command.

```
export AWS_DEFAULT_REGION=us-west-2
export AWS_ACCESS_KEY_ID=<filled out id>
export AWS_SECRET_ACCESS_KEY=<filled out key>
export AWS_SESSION_TOKEN=<filled out token>

fiboa publish fi -e finland -c ~/Downloads fi_data
```

Reply:
I added some documentation / user instructions...